src/components/service-room.ts

```diff
@@ -0,0 +1,167 @@
+import { MatrixClient } from "matrix-bot-sdk";
+
+export enum ServiceNotificationNoticeCode {
+	Unknown = "UNKNOWN",
+	Blocked = "BLOCKED",
+	RemoteServiceOutage = "REMOTE_SERVICE_OUTAGE",
+	MatrixServiceOutage = "MATRIX_SERVICE_OUTAGE",
+}
+
+export enum ServiceNotificationServerity {
+	/**
+	 * Just information for the administrator, usually no action required.
+	 */
+	Infomational = "info",
+	/**
+	 * Something the administrator should know about, might require an explicit notification.
+	 */
+	Warning = "warning",
+	/**
+	 * A serious issue has occured with the bridge, action needed.
+	 */
+	Error = "error",
+	/**
+	 * The bridge cannot function, action urgently needed.
+	 */
+	Critical = "critical"
+}
+
+
+export interface ServiceRoomOpts {
+	/**
+	 * The roomId to send notices to.
+	 */
+	roomId: string;
+	/**
+	 * The minimum time allowed before
+	 * a new notice with the same ID can be sent (to avoid room spam).
+	 * Defaults to a hour.
+	 */
+	minimumUpdatePeriodMs?: number;
+	/**
+	 * The prefix to use in state keys to uniquely namespace the bridge.
+	 */
+	bridgeStateKeyPrefix: string;
+	/**
+	 * Any metadata to be included in all notice events.
+	 */
+	metadata: Record<string, unknown>
```

Comment:
Would it help to define a type alias for this?  It could be shared by both this and `NotificationEventContent.metadata`.